From 21b0ea02857bdddd4da59b7193fd07a2338c4843 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Mon, 26 Jun 2017 14:16:33 +0800
Subject: [PATCH] drm: record hdr_source_metadata_property blob_id

The blob_id is used to get blob data of hdr_source_metadata_property.

Change-Id: I8383265e50261438cf8484bd731e42fc23c24c51
Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/gpu/drm/drm_atomic.c | 2 ++
 include/drm/drm_crtc.h       | 1 +
 2 files changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/drm_atomic.c b/drivers/gpu/drm/drm_atomic.c
index 94ec204ce87b..9f80708e3a77 100644
--- a/drivers/gpu/drm/drm_atomic.c
+++ b/drivers/gpu/drm/drm_atomic.c
@@ -1010,6 +1010,8 @@ int drm_atomic_connector_set_property(struct drm_connector *connector,
 				-1,
 				&replaced);
 		state->hdr_metadata_changed |= replaced;
+		if (replaced)
+			state->blob_id = val;
 		return ret;
 	} else if (connector->funcs->atomic_set_property) {
 		return connector->funcs->atomic_set_property(connector,
diff --git a/include/drm/drm_crtc.h b/include/drm/drm_crtc.h
index 75e7abcf5810..8e483c28b116 100644
--- a/include/drm/drm_crtc.h
+++ b/include/drm/drm_crtc.h
@@ -684,6 +684,7 @@ struct drm_connector_state {
 	 */
 	struct drm_property_blob *hdr_source_metadata_blob_ptr;
 	bool hdr_metadata_changed : 1;
+	uint64_t blob_id;
 };
 
 /**
-- 
2.35.3

